### DISCUSSION TITLE
Winget Upgrade

### DISCUSSION THREAD

**Discussion by knevan** [microsoft/winget-cli#5043] [General]
Will winget upgrade --all instantly upgrade everything by asking for a Y or N confirmation? And why is my winget not upgraded automatically?

**Comment by davidldennison** [comment]
Depends on your configuration but normally it will run each installer one after the other until they have all been upgraded.

**Reply by knevan** [reply]
How to configure to ask for confirmation before upgrading -all? 🙏

**Comment by denelon** [comment]
Running `winget upgrade --all` tells WinGet to find any installed program with a newer version available in any of the configured sources, and to upgrade them.

If you run `winget upgrade` without the "--all" you can see the list of available updates.

You can then decide which ones you want to upgrade and run them individually like `winget upgrade <packageId>`.

Depending on what you're trying to achieve, there may be other commands, arguments, or settings to achieve the desired outcome.

**Reply by davidldennison** [reply]
I'm not sure if this is going out on a limb here, but does this imply that `winget upgrade --all` uses the Winget COM API differently than just `winget upgrade`?
